cmd/cluster/from_infra_id.go

```diff
@@ -0,0 +1,107 @@
+package cluster
+
+import (
+	"encoding/json"
+	"fmt"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/osdctl/internal/utils/globalflags"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"os"
+	"strings"
+)
+
+type fromInfraIdOptions struct {
+	globalOpts *globalflags.GlobalOptions
+}
+
+type clusterIdentification struct {
+	ID         string `json:"id" yaml:"ID"`
+	ExternalID string `json:"external_id" yaml:"ExternalID"`
+	Name       string `json:"name" yaml:"Name"`
+}
+
+func (ci clusterIdentification) String() string {
+	return fmt.Sprintf("ID:\t\t%s\nExternal ID:\t%s\nName:\t\t%s\n", ci.ID, ci.ExternalID, ci.Name)
+}
+
+func newCmdFromInfraId(globalOpts *globalflags.GlobalOptions) *cobra.Command {
+	opts := &fromInfraIdOptions{
+		globalOpts,
+	}
+	return &cobra.Command{
+		Use:               "from-infra-id",
+		Short:             "Get cluster ID and external ID from a given infrastructure ID commonly used by Splunk",
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(opts.complete(cmd, args))
+			cmdutil.CheckErr(opts.run(cmd, args))
+		},
+	}
+}
+
+func (ops *fromInfraIdOptions) complete(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
```

Comment:
It's not hurting anything, but this if statement shouldn't be needed as you're ensuring `cobra.ExactArgs(1)` on the command definition.

Reply:
Good call.

Reply:
Fixed.